### PR TITLE
Add Kademlia article

### PR DESCRIPTION
**Pull request by danielrachi1**
I wrote this article several weeks ago when I was starting to learn about The Portal Network. I realized that it would be a good addition to the portal docs.

Originally, I thought about putting this in the Trin book, ethereum/trin#922, but after talking with the Trin team it was decided that this would be a better place for that article.

I also included some existing info that was available [on the trin book](https://ethereum.github.io/trin/developers/protocols/kademlia.html).

### To Do

- [x] Have someone fact-check the article.
- [x] Make sure that the folder structure I used is correct, and that the article is in the right spot on the website.

### CODE REVIEW

**Comment by jmcook1186** [comment]
Hi - only just saw this PR - looks awesome, thanks!

**Review by jmcook1186 — APPROVED** [review]

File: pages/resources/protocols/kademlia.mdx [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Kademlia Distributed hash table
```
Comment (njgheorghita) [inline pages/resources/protocols/kademlia.mdx]: *nit* `Kademlia DHT (Distributed Hash Table)`

**Review by njgheorghita — APPROVED** [review]
Nice! @danielrachi If you don't mind also opening a pr in `trin` that deletes this content from our book that would be 💯

**Comment by danielrachi1** [comment]
@jmcook1186 pls let me know if you want me to squash bot commits together :)

**Pull request merged**